### PR TITLE
Update GreyNoise command to allow use of Community API

### PR DESCRIPTION
**Pull request by bradchiappetta**
Included are some suggested updates to help to integrate the GreyNoise Community API into Harpoon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ After configuration the following plugins are available within the `harpoon` com
     fullcontact         Requests Full Contact API (https://www.fullcontact.com/)
     github              Request Github information through the API
     googl               Requests Google url shortener API
-    greynoise           Request Grey Noise API
+    greynoise           Request information from GreyNoise API (pick Community or Enterprise via api_type config)
```
Comment (bradchiappetta) [inline README.md:73]: Just adding some clarity about using either API version.

File: README.md [hunk 2]
```diff
@@ -124,7 +124,7 @@ You can get information on each command with `harpoon help COMMAND`
 * [CIRCL Passive DNS](https://www.circl.lu/services/passive-dns/)
 * [Farsight Dnsdb](https://www.farsightsecurity.com/dnsdb-community-edition/)
 * [FullContact](https://dashboard.fullcontact.com/register)
-* [GreyNoise](https://greynoise.io/)
+* [GreyNoise](https://viz.greynoise.io/account) : supports both Community and Enterprise API.  Use api_type config setting to specify which API type to use.  Both still require an API key to work.
```
Comment (bradchiappetta) [inline README.md:127]: Some extra details about where to get GN API key and how to define with API to use in the conf.

File: harpoon/commands/gn.py [hunk 3]
```diff
@@ -16,16 +15,19 @@ class CommandGreyNoise(Command):
...
     * List tags: `harpoon greynoise -l` (default output as json)
     * Search for an IP: `harpoon greynoise -i IP`
     * Run a GNQL query: `harpoon greynoise -q "classification:malicious tags:'emotet'"`
     """
 
     name = "greynoise"
-    description = "Request Grey Noise API"
-    config = {"GreyNoise": ["key"]}
+    description = "Request information from GreyNoise API"
+    config = {"GreyNoise": ["key", "api_type"]}
```
Comment (bradchiappetta) [inline harpoon/commands/gn.py:30]: Added a new input to specify if harpoon should use the GN Paid or Community API.

File: harpoon/commands/gn.py [hunk 4]
```diff
@@ -52,30 +54,27 @@ def print_results(self, res, args):
...
-                return r.json()["tags"]
-            else:
-                raise GreynoiseError(e)
-        except Exception as e:
-            raise GreynoiseError(e)
-
     def run(self, conf, args, plugins):
         logging.getLogger("greynoise").setLevel(logging.CRITICAL)
-        gn = GreyNoise(api_key=conf["GreyNoise"]["key"])
+        if conf["GreyNoise"]["api_type"].lower() == "community":
```
Comment (bradchiappetta) [inline harpoon/commands/gn.py:59]: Allows selection of API via the conf parameter api_type

File: harpoon/commands/gn.py [hunk 5]
```diff
@@ -52,30 +54,27 @@ def print_results(self, res, args):
...
+            gn = GreyNoise(
+                api_key=conf["GreyNoise"]["key"],
+                integration_name="Harpoon (https://github.com/Te-k/harpoon)",
+                offering="community",
+            )
+        else:
+            gn = GreyNoise(
+                api_key=conf["GreyNoise"]["key"],
+                integration_name="Harpoon (https://github.com/Te-k/harpoon)",
```
Comment (bradchiappetta) [inline harpoon/commands/gn.py:68]: Wanted to ensure your UA string was being passed to the SDK correctly.

File: harpoon/commands/gn.py [hunk 6]
```diff
@@ -52,30 +54,27 @@ def print_results(self, res, args):
...
+            )
         if args.ip:
             res = gn.ip(args.ip)
             self.print_results(res, args)
         elif args.query:
             res = gn.query(args.query)
             self.print_results(res, args)
         elif args.list:
-            res = self.get_list_tags()
+            res = gn.metadata()
```
Comment (bradchiappetta) [inline harpoon/commands/gn.py:77]: This is the updated SDK command to get all tags.

**Comment by Te-k** [comment]
Hi @bradchiappetta 
This is awesome. Thanks for the work!

**Pull request merged**